pkgs/development/python-modules/pytest/4.nix

```diff
@@ -25,7 +20,14 @@ buildPythonPackage rec {
   doCheck = !isPyPy; # https://github.com/pytest-dev/pytest/issues/3460
   checkPhase = ''
     runHook preCheck
-    $out/bin/py.test -x testing/ -k "not test_collect_pyargs_with_testpaths"
+
+    # don't test bash builtins
+    rm testing/test_argcomplete.py
+
+    # determinism - this test writes non deterministic bytecode
+    rm -rf testing/test_assertrewrite.py
+
+    PYTHONDONTWRITEBYTECODE=1 $out/bin/py.test -x testing/ -k "not test_collect_pyargs_with_testpaths"
```

Comment:
A lot of files have indeed been rewritten with this bad bytecode.

Reply:
should we add `PYTHONDONTWRITEBYTECODE=1` to the pytestCheckHook @FRidh ?

Reply:
Maybe. It can happen in case `pytest` is simply used. A better solution is probably for it to adhere to `SOURCE_DATE_EPOCH`, but there's probably a reason why this custom function exists.